util/upload-cookbook.py

```diff
@@ -0,0 +1,241 @@
+import re
+from datetime import datetime
+
+import argparse
+import boto3
+from botocore.exceptions import ClientError
+
+_BACKUP_DIR = "backup"
+_COOKBOOKS_DIR = "cookbooks"
+_bck_date = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+_bck_error_array = set()
+_cp_error_array = set()
+_ls_error_array = set()
+credentials = []
+
+
+def get_all_aws_regions(region):
+    ec2 = boto3.client("ec2", region_name=region)
+    return set(sorted(r.get("RegionName") for r in ec2.describe_regions().get("Regions")))
+
+
+def _aws_ls(s3, region, bucket_name, key):
+    out = s3.list_objects_v2(Bucket=bucket_name, Prefix=key)
+    if len(out.get("Contents", [])) > 0:
+        _ls_error_array.add(region)
+
+
+def put_object_to_s3(s3_client, bucket, folder, full_name):
+    key = folder + "/" + full_name
+    try:
+        s3_client.upload_file(full_name, bucket, key, ExtraArgs={"ACL": "public-read"})
+
+        print("Successfully uploaded {0} to s3://{1}/{2}".format(full_name, bucket, key))
+    except ClientError as e:
+        print("Couldn't upload {0} to bucket s3://{1}/{2}".format(full_name, bucket, key))
+        if e.response["Error"]["Code"] == "NoSuchBucket":
+            print("Bucket is not present.")
+
+        raise e
+
+
+def _aws_bck(s3, args, region, bucket_name, full_name):
+    if args.dryrun:
+        print(
+            "Not backing up {0} to bucket {1} override is {2}, dryrun is {3}".format(
+                full_name, bucket_name, args.override, args.dryrun
+            )
+        )
+    else:
+        try:
+            copy_source = {"Bucket": bucket_name, "Key": _COOKBOOKS_DIR + "/" + full_name}
+            s3.copy(copy_source, bucket_name, _BACKUP_DIR + "/" + full_name + _bck_date)
+        except ClientError as e:
+            print("Couldn't backup {0}".format(full_name))
+            if e.response["Error"]["Code"] == "NoSuchBucket":
+                print("Bucket is not present.")
+            _bck_error_array.add(region)
+
+
+def _aws_cp(s3, args, region, bucket_name, folder, full_name):
+    if args.dryrun:
+        print(
+            "Not uploading {0} to bucket {1}, override is {2}, dryrun is {3}".format(
+                full_name, bucket_name, args.override, args.dryrun
+            )
+        )
+    else:
+        key = folder + "/" + full_name
+        try:
+            s3.upload_file(full_name, bucket_name, key, ExtraArgs={"ACL": "public-read"})
```

Comment:
this code seems the same of: put_object_to_s3

Could we put it in common?

Reply:
put_object_to_s3 was a copy/paste from elsewhere, not used here. Removing it